common-app-lib/src/cmr/common_app/api/request_logger.clj

```diff
@@ -116,55 +136,60 @@
 ;; time
 
 (defn log-ring-request
-  "Log a request from Ring returning JSON data to be parsed by a log tool like
-   Splunk or ELK. This handler can be called multiple times at different positions
-   in the handlers call. When calling this function multiple times in the routes
-   list, then pass in an ID (the two arity call) and the output will include the
-   ID value in the log so that log entries can be told apart. One may want to do
-   this to see when a value logged becomes valid in the routes chain and thus
-   availible downstream.
-
-   NOTE: If you pass in an ID of :ignore then no ID will be written out, the
-   same behavior if you call the single arity call."
+  "Log a request from Ring returning JSON data to be parsed by a log tool like Splunk or ELK. This
+   handler can be called multiple times at different positions in the handlers call but it does
+   depend on common-routes/add-request-id-response-handler and add-time-stamp. When calling this
+   function multiple times in the routes list, then pass in an ID (the two arity call) and the
+   output will include the ID value in the log so that log entries can be told apart. One may want
+   to do this to see when a value logged becomes valid in the routes chain and thus availible
+   downstream.
+
+   NOTE: If you pass in an ID of :ignore then no ID will be written out, the same behavior if you
+   call the single arity call."
   ([handler]
    (log-ring-request handler :ignore-id))
   ([handler id]
    (fn [request]
      (if-not (config/enable-enhanced-http-logging)
        (handler request)
-       (let [start (tk/now-ms)
+       (let [response (handler request) ;; Do all the response handlers first
+             log-start (tk/now-ms) ;; After processing the other handlers, start tracking log time
+             _ (when-not (:ring-start-time request) ;; Did someone forget to setup a routes.clj
+                 (error "There is no ring start time for this service" (request->uri request)))
```

Comment:
when configured correctly, this error should never happen, but if someone changes the ring order, this message will be issued if the order results in a failed setup.